tests/core/gui/test_main_window.py

```diff
@@ -208,96 +226,103 @@ def _folder_ready():
+            assert f_w.table_files.rowCount() == 2
+            folder = f_w.table_files.item(1, 1)
+            assert folder
+            assert folder.text() == "dir1"
+        else:
+            assert f_w.label_role.text() == _(WorkspaceRole.READER)
+            assert isinstance(f_w.workspace_fs, WorkspaceFSTimestamped)
 
-        await aqtbot.wait_until(lambda: _folder_ready(False))
+    await aqtbot.wait_until(lambda: _folder_ready(False))
 
-        assert logged_gui.tab_center.count() == 1
+    assert logged_gui.tab_center.count() == 1
 
-        # Generate a file link
-        url = f_w.workspace_fs.generate_file_link(f_w.current_directory, timestamp)
+    # Generate a file link
+    url = f_w.workspace_fs.generate_file_link(f_w.current_directory, timestamp)
 
-        # Unmount the original workspace
-        await core.mountpoint_manager.unmount_workspace(f_w.workspace_fs.workspace_id, None)
+    # Unmount the original workspace
+    await core.mountpoint_manager.unmount_workspace(f_w.workspace_fs.workspace_id, None)
 
-        def _unmounted(ts):
-            assert not core.mountpoint_manager.is_workspace_mounted(
-                f_w.workspace_fs.workspace_id, ts
-            )
+    def _unmounted(ts):
+        assert not core.mountpoint_manager.is_workspace_mounted(f_w.workspace_fs.workspace_id, ts)
 
-        # Making sure that it is unmounted
-        await aqtbot.wait_until(lambda: _unmounted(None))
+    # Making sure that it is unmounted
+    await aqtbot.wait_until(lambda: _unmounted(None))
 
-        # Add an instance with the file link
-        logged_gui.add_instance(url.to_url())
+    # Add an instance with the file link
+    logged_gui.add_instance(url.to_url())
 
-        # Workspace should be mounted
-        await aqtbot.wait_until(lambda: _mounted(timestamp))
+    # Workspace should be mounted
+    await aqtbot.wait_until(lambda: _mounted(timestamp))
 
-        # If the link was created with a timestamp, the workspace should be a
-        # TimestampedWorkspace, otherwise it's just a normal workspace
-        await aqtbot.wait_until(lambda: _folder_ready(timestamp is not None))
+    # If the link was created with a timestamp, the workspace should be a
+    # TimestampedWorkspace, otherwise it's just a normal workspace
+    await aqtbot.wait_until(lambda: _folder_ready(timestamp is not None))
 
 
 @pytest.mark.gui
 @pytest.mark.trio
-@pytest.mark.parametrize("timestamp", [True, False])
-async def test_link_file_invalid_path(aqtbot, autoclose_dialog, logged_gui_with_files, timestamp):
+@pytest.mark.parametrize(
+    "toggle_timestamp", [True, False], ids=["With Timestamp", "Without Timestamp"]
+)
+async def test_link_file_invalid_path(
+    aqtbot,
+    autoclose_dialog,
+    logged_gui_with_files: LoggedGuiWithFiles,
+    toggle_timestamp: bool,
+):
     logged_gui, w_w, f_w = logged_gui_with_files
     await f_w.workspace_fs.sync()
 
-    with freeze_time(DateTime.now().add(seconds=1)):
```

Comment:
Why ?

Reply:
Because I can that's why :wink: 

That simplify the test, here the `freeze_time` serve no purpose

Reply:
What did you change to solve the timeout ?

Reply:
nothing :shrug: I can't reproduce it